core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -225,6 +242,20 @@ public void resetContext() throws Exception {
     verify(mMockFileSystemContext).reset(alluxio.Configuration.global());
   }
 
+  /**
+   * Tests that initializing the {@link AbstractFileSystem} with Alluxio on Zookeeper URI
+   * will reinitialize the file system context.
+   */
+  @Test
+  public void resetContextWithZkUri() throws Exception {
+    // Change to otherhost:410
```

Comment:
there are many cases to test. It isn't clear which one this test covers

non-zk to zk
zk to different zk
zk to same zk
zk to non-zk

Reply:
The mock file system context havs set default master address to be "defaultHost".
```
    when(mMockMasterInquireClient.getConnectDetails()).thenReturn(
        new SingleMasterConnectDetails(new InetSocketAddress("defaultHost", 1)));
    when(mMockFileSystemContext.getMasterAddress())
        .thenReturn(new InetSocketAddress("defaultHost", 1));
```

So here to change from non-zk to zk.

I also added some other tests
non-zk to non-zk
```resetContext() ```
non-zk to zk
```
resetContextUsingSingleZkUri() 
resetContextUsingMultipleZkUri() 
resetContextUsingMultipleZkUriWithSemicolons
```
and some more complicated situation:
```
resetContextFromSingleZkUriToAnotherSingleZkUri
resetContextFromSingleZkUriToMultipleZkUri
resetContextFromZkUriToNonZkUri
```